client/cli/go/cmds/device.go

```diff
@@ -322,30 +324,83 @@ var deviceDisableCommand = &cobra.Command{
 }
 
 var deviceResyncCommand = &cobra.Command{
-	Use:     "resync [device_id]",
+	Use:     "resync [object_id]",
 	Short:   "Resync storage information about the device with operation system",
 	Long:    "Resync storage information about the device with operation system",
-	Example: "  $ heketi-cli device resync 886a86a868711bef83001",
+	Example: `  $ heketi-cli device resync 886a86a868711bef83001 `,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		//ensure proper number of args
 		s := cmd.Flags().Args()
+
+		//ensure proper number of args
 		if len(s) < 1 {
-			return errors.New("device id missing")
+			return errors.New("object/cluster id missing")
 		}
 
-		// Set node id
-		deviceId := cmd.Flags().Arg(0)
+		// set objectId
+		objectId := cmd.Flags().Arg(0)
+		// determine operation mode: CLUSTER od DEVICE
+		resyncCluster, err := cmd.Flags().GetBool("cluster")
+		if err != nil {
+			return err
+		}
 
 		// Create a client
 		heketi, err := newHeketiClient()
 		if err != nil {
 			return err
 		}
 
-		//set url
-		err = heketi.DeviceResync(deviceId)
-		if err == nil {
-			fmt.Fprintf(stdout, "Device %v updated\n", deviceId)
+		// a single device Id was specified and no cluster resync was requested
+		if resyncCluster == false {
+			fmt.Println("No cluster-wide resync operation requested. Using object_id as device_id.")
+			//set url
+			err = heketi.DeviceResync(objectId)
+			if err == nil {
+				fmt.Fprintf(stdout, "Device %v updated\n", objectId)
+			}
+		} else { // requested cluster resync..
+			// RESYNC ALL
+			fmt.Println("Cluster-wide resync operation requested. Using object_id as cluster_id.")
+			// Get Topology Info from Server
+			topoinfo, err := heketi.TopologyInfo()
+			if err != nil {
+				return err
+			}
+
+			// Iterate over Clusters
+			for cluster, _ := range topoinfo.ClusterList {
+				thisCluster := topoinfo.ClusterList[cluster]
+				// check cluster id.
+				if thisCluster.Id == objectId {
+					// iterate over Nodes
+					for node, _ := range thisCluster.Nodes {
+						thisNode := thisCluster.Nodes[node]
+						// resync every Device in this Node
+						for _, device := range thisNode.DevicesInfo {
+							fmt.Fprintf(stdout, "\t\tDevice Id:%-35v"+
```

Comment:
I'm not sold on printing all of this state out before the device resync is called. Why not just print something like ("Resyncing device %v...", device.Id)

Reply:
Yeah, I kind of liked some verbose printout but I see that it clobbers the output.
I will change that